### PR TITLE
add deprecation warnings for server extensibility

### PR DESCRIPTION
**Pull request by ankithkonda**
Server extensibility is being deprecated. We should update the docs

In all the pages under extensibility, we now have - 
Screenshot 2022-11-14 at 1 05 54 pm

### CODE REVIEW

**Review by harrisonmeister — COMMENTED** [review]
Made some minor suggestions, mainly grammarly edits :smile:. A couple of things you could optionally consider:

 - The smaller warning is repeated several times in the header for pages. You could make it an include and re-use that text once.
 - Should this warning also be added to this new [deprecations page](https://octopus.com/docs/deprecations)? See https://github.com/OctopusDeploy/docs/pull/1716 as an example where deprecations have recently been added for 2022.3

File: docs/administration/server-extensibility/installing-a-custom-server-extension.md [hunk 1]
```diff
@@ -3,6 +3,29 @@ title: Installing a custom server extension
...
+|**Guest**|`OCTOPUS__FeatureToggles__GuestAuthenticationProviderFeatureToggle`|
+|**UsernamePassword**|`OCTOPUS__FeatureToggles__UsernamePasswordAuthenticationProviderFeatureToggle`|
+|**DirectoryServices**|`OCTOPUS__FeatureToggles__DirectoryServicesAuthenticationProviderFeatureToggle`|
+|**LDAP**|`OCTOPUS__FeatureToggles__LdapAuthenticationProviderFeatureToggle`|
+|**AzureAD**|`OCTOPUS__FeatureToggles__AzureADAuthenticationProviderFeatureToggle`|
+|**GoogleApps**|`OCTOPUS__FeatureToggles__GoogleAppsAuthenticationProviderFeatureToggle`|
+|**Okta**|`OCTOPUS__FeatureToggles__OktaAuthenticationProviderFeatureToggle`|
+|**OctopusID**|`OCTOPUS__FeatureToggles__OctopusIDAuthenticationProviderFeatureToggle`|
+:::
+
```
[the remark this answers is not in the document]
Comment (ankithkonda) [reply docs/administration/server-extensibility/installing-a-custom-server-extension.md]: We no longer require env variables to be set, removing this part

**Comment by ankithkonda** [comment]
> You could make it an include and re-use that text once

@harrisonmeister That's neat. Is there an example?

Sorry, this gathered dust for a couple of months. I made the suggested changes and also added to the deprecations page. Could you please have another look 🙇

File: docs/deprecations/index.md [hunk 2]
```diff
@@ -22,3 +22,6 @@ Deprecations are subject to change in detail or timeframe. If you need help asse
 
 * The Space level `/useronboarding` API endpoint is being removed in future versions of Octopus. It was used internally to improve the user onboarding experience. We have since reworked the new user experience and removed the old endpoint. There is no replacement for this endpoint. We do not expect that anyone outside our internal teams has used this endpoint. If you believe this could negatively affect you, please get in touch with our [support team](https://octopus.com/support).
 
+## Server Extensbility
```
Comment (harrisonmeister) [inline docs/deprecations/index.md:25]: @ankithkonda - Is there a version this is planned to be deprecated from? Looking at the existing titles that are H2 headings in this page, it looks like it's designed to be done by the Server version following a heading convention like this:

```md
## Deprecations for 20xx.X
```

**Comment by harrisonmeister** [comment]
> > You could make it an include and re-use that text once
> 
> @harrisonmeister That's neat. Is there an example?
> 
> Sorry, this gathered dust for a couple of months. I made the suggested changes and also added to the deprecations page. Could you please have another look bow

Take a look at this commit https://github.com/OctopusDeploy/docs/pull/1720/commits/c05b54da687bdb33d547b8e9d2607ac4d167ca86 - it should show what I was thinking (no changes are required to it, if you're happy with the content) :smile:

File: docs/deprecations/index.md (hunk 2 above)
Comment (ankithkonda) [reply docs/deprecations/index.md:25]: I would say 2022.4 is when we deprecated it (as part of auth consolidation) and are looking for customers to reach out to us, with end of 2023 being when we expect to remove the functionality. 

I can put this under 2022.4 title 👍

Comment (ankithkonda) [reply docs/deprecations/index.md]: I've just pushed up an update to the title :)

Comment (harrisonmeister) [reply docs/deprecations/index.md]: Looks good, thanks

**Review by harrisonmeister — APPROVED** [review]
LGTM

**Pull request merged**